src/entries-list/node-card.ts

```diff
@@ -0,0 +1,266 @@
+
+  @query("mwc-dialog") private _deleteDialog!: Dialog;
+
+  protected render() {
+    return html`
+      <esphome-card
+        class=${classMap({
+          "status-online": this.onlineStatus === true,
+          "status-offline": this.onlineStatus === false,
+          "status-unknown":
+            this.onlineStatus === null || this.onlineStatus === undefined,
+          highlighted: this.highlighted,
+        })}
+      >
+        <div class="card-header">
+          ${this.entry.name}
+          ${"web_server" in this.entry.loaded_integrations
+            ? html`
+                <div class="tooltip-container">
+                  <a href=${`http://${this.entry.address}`} target="_blank">
+                    <mwc-icon>launch</mwc-icon>
+                  </a>
+                  <paper-tooltip>Open Node Web Server Interface</paper-tooltip>
+                </div>
+              `
+            : ""}
+          ${this.entry.deployed_version != this.entry.current_version
+            ? html`
+                <div class="tooltip-container">
+                  <mwc-icon class="update-available">system_update</mwc-icon>
+                  <paper-tooltip
+                    >Update Available: ${this.entry.deployed_version}
+                    &#x27A1;&#xFE0F;${this.entry.current_version}</paper-tooltip
+                  >
+                </div>
+              `
+            : ""}
+
+          <esphome-button-menu
+            corner="TOP_START"
+            @action=${this._handleOverflowAction}
+          >
+            <mwc-icon-button slot="trigger" icon="more_vert"></mwc-icon-button>
+            <mwc-list-item>Clean Build Files</mwc-list-item>
+            <mwc-list-item>Delete</mwc-list-item>
+            ${"mqtt" in this.entry.loaded_integrations
+              ? html`<mwc-list-item>Clean MQTT</mwc-list-item>`
+              : ""}
+          </esphome-button-menu>
+        </div>
+
+        <div class="card-content">
+          <div class="node-config-path tooltip-container">
+            <code class="inlinecode">${this.entry.filename}</code>
+            <paper-tooltip>
+              Full Path: <code class="inlinecode">${this.entry.path}</code>
+            </paper-tooltip>
+          </div>
+
+          <div class="online-status">
+            <span class="indicator"></span>
+          </div>
+
+          ${this.entry.comment
+            ? html`<div class="node-card-comment">${this.entry.comment}</div>`
+            : ""}
+        </div>
+
+        <div class="card-actions">
+          <mwc-button label="Edit" @click=${this._handleEdit}></mwc-button>
+          <mwc-button
+            label="Validate"
+            @click=${this._handleValidate}
+          ></mwc-button>
+          <mwc-button
+            label="Install"
+            @click=${this._handleInstall}
+          ></mwc-button>
+          <mwc-button label="Logs" @click=${this._handleLogs}></mwc-button>
+        </div>
+      </esphome-card>
+
+      <mwc-dialog>
```

Comment:
Let's consider moving this to it's own component, like we do with other dialogs.